manifest/docker_schema2.go

```diff
@@ -207,7 +207,14 @@ func (m *Schema2) UpdateLayerInfos(layerInfos []types.BlobInfo) error {
 	original := m.LayersDescriptors
 	m.LayersDescriptors = make([]Schema2Descriptor, len(layerInfos))
 	for i, info := range layerInfos {
-		m.LayersDescriptors[i].MediaType = original[i].MediaType
+		switch info.Compression {
+		case types.PreserveOriginal:
+			m.LayersDescriptors[i].MediaType = original[i].MediaType
+		case types.Decompress:
+			m.LayersDescriptors[i].MediaType = DockerV2SchemaLayerMediaTypeUncompressed
+		case types.Compress:
+			m.LayersDescriptors[i].MediaType = DockerV2SchemaLayerMediaType
+		}
```

Comment:
Yes, this implementation approach accounts for the different MIME types in different manifest formats, but this will also soon need to handle different compression algorithms (gzip vs. Zstd).  Very broadly like `compression.Algorithm` in #639 , but that does not seem to provide constants for known formats yet.

Reply:
@mtrmac @giuseppe, so once #639 is merged, we need to add a new MIME type for `zstd` (e.g., `application/vnd.docker.image.rootfs.diff.tar.zstd` and `application/vnd.oci.image.layer.v1.tar+zstd`), right?

Reply:
Well, that, and `BlobInfo.Compression` will either need to be replaced or augmented to also record the algorithm.